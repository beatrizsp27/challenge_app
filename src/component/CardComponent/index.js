import React from 'react';

const CardComponent = () => {
	return <h1>Hola mundo MainDetailsProduct </h1>;
};

export default CardComponent;
