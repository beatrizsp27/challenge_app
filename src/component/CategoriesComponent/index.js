import React from "react";

const CategoriesComponent = ({categoriesArray}) =>{
    return(
        <div className={'screens_component_category'}>
            {
                categoriesArray && categoriesArray.length > 0 && (
                    <>
                        <div  className={'screens_category'}>
                            <div  className={'list_ul'}>
                                {categoriesArray.map((categoryItem, index)=> {
                                    const categoriesArrayLength = categoriesArray.length;
                                    const isOtherFormat = index+1 === categoriesArrayLength;
                                    const sign = ' >'
                                    return (
                                        // eslint-disable-next-line react/jsx-key
                                        <li key={index} >
                                            <span  className={isOtherFormat? 'screens_title_body_header_bold' :'screens_title_body_header'}> {isOtherFormat ? categoryItem : `${categoryItem}${sign}`} </span>
                                        </li>
                                    )})}
                            </div>
                        </div>
                    </>
                )
            }
        </div>
    );
};

export default CategoriesComponent;
