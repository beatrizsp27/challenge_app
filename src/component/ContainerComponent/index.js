import React from 'react';

const ContainerComponent = ({ children }) => {
    return <div className='screens_component_div'>{children}</div>;
};

export default ContainerComponent;
