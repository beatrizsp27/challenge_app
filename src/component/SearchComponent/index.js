import React from 'react';
import {logo, search} from "../../assets";
import s from "../../strings";

const SearchComponent = ({ handleChange, onAction }) => {
    return (
        <div className={'screens_details_header'}>
            <form onSubmit={onAction}>
                <div className={'screens_details_form'}>
                    <div className={'box_img_search'}>
                        <img className={'box_img'} src={logo}/>
                    </div>
                    <div className={'box_input_icon_search'} >
                        <input
                            autoComplete={'false'}
                            name={'search'}
                            className="box_input_text"
                            type="search"
                            placeholder={s.search}
                            onChange={handleChange}
                        />
                        <div className={'box_search_icon'} onClick={onAction}>
                            <img  className={'box_img_mel'}  src={search} alt={''}/>
                        </div>
                    </div>
              </div>
            </form>
        </div>
    )
};

export default SearchComponent;
