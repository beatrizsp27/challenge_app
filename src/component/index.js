export { default as ContainerComponent } from './ContainerComponent';
export { default as SearchComponent } from './SearchComponent';
