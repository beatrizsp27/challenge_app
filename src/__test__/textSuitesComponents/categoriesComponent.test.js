import React from "react";
import { CategoriesComponent } from '../../component';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

/** SE CONFIGURA EL ADAPTADOR**/
import {shallow, configure} from "enzyme";
configure({ adapter: new Adapter() })

describe("CategoriesComponent", ()=>{

    /** VARIABLES**/
    let wrapperCategorias;
    const categoriesArray = ["ropa", 'electronico'];

    beforeAll(()=>{
        /** SE MONTA EL COMPONENTE**/
        wrapperCategorias = shallow(<CategoriesComponent categoriesArray={categoriesArray} />);
    });

    /** SE VALIDA QUE EL COMPONENTE ESTE DEFENIDO Y QUE EXISTA**/
    test("renderear correctamente", () => {
        expect(wrapperCategorias).toBeTruthy()
        wrapperCategorias.find('div');
        expect(wrapperCategorias.length).toEqual(1);
    })


    /** SE GENERA EL SNAPSHAP PARA VALIDAR LOS QUE LOS CAMBIOS DEL COMPONENTE SEAN CONTROLADOS**/
    test("Se crea el snap despues de las pruebas", () => {
        const treeSearch = renderer.create(<CategoriesComponent />).toJSON();
        expect(treeSearch).toMatchSnapshot();
    })
});
