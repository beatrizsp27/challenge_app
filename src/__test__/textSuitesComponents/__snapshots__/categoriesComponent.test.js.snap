// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrapperCategorias Se crea el snap despues de las pruebas 1`] = `
<div
  className="screens_component_category"
/>
`;
